### PR TITLE
Add hostname configuration docs

### PR DESCRIPTION
**Pull request by fgiudici**
Added a section to document how the hostname configuration works.
Another PR will follow on how to set the hostname from DHCP (for the Next section only).

Related to https://github.com/rancher/elemental-docs/issues/198 .

### CODE REVIEW

File: docs/hostname.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Elemental Operator, which creates a MachineInventory for the host.
+
+Each host registered with the Elemental Operator is tracked by a `MachineInventory` resource.
+The hostname of the host is eventually set to the `name` of the `MachineInventory` resource,
+during the Cluster provisioning phase (i.e., during k3s/RKE2 deployment on the host).
+
+#### Default hostname
+
+The default name assigned to the newly created MachineInventory is in the form `m-{$UUID}`.
+When the host is provisioned as part of a Cluster, the `m-{UUID}` hostname is set, overriding the previous `rancher-{$RANDOM}` hostname.
```
Comment (anmazzotti) [inline docs/hostname.md:23]: Shouldn't this be different for the next release?
I thought `rancher-{$RANDOM}` was removed with a recent PR.

I also miss the reference on how not to set the default hostname (and using the DHCP provided one).

File: docs/hostname.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+title: ''
+---
+
+<head>
+  <link rel="canonical" href="https://elemental.docs.rancher.com/hostname"/>
+</head>
+
+### How to set a custom hostname
+
+When booting a host with an Elemental ISO, the hostname is temporarily set to `rancher-{$RANDOM}`.
```
Comment (anmazzotti) [inline docs/hostname.md:12]: Same here as the other comment. 
And the PR I mentioned is this one: https://github.com/rancher/elemental/pull/1171/files

**Review by kkaempf — APPROVED** [review]
lgtm

File: docs/hostname.md (hunk 2 above)
Comment (fgiudici) [reply docs/hostname.md:12]: The plan was to first document the actual state and how the hostname is managed, then update the next section.
For the next update I want to wait https://github.com/rancher/elemental/pull/1171 or some other change allowing hostname from DHCP is merged.
Before merging 1171 would like to try another approach  (i.e., set a transient hostname).

**Review by anmazzotti — APPROVED** [review]

**Pull request merged**